internal/git/gitsync_processor.go

```diff
@@ -76,56 +78,107 @@ func watchRepo(ctx context.Context, restConfig *rest.Config, repo *v1.Repository
 	}
 
 	// The revision can be a branch, a tag, or a commit hash
-	h, err := r.ResolveRevision(plumbing.Revision(repo.TargetRevision))
+	hash, err := r.ResolveRevision(plumbing.Revision(repo.TargetRevision))
```

Comment:
say somebody has a bad `TargetRevision` specified - we will be in a tight loop of retrying over and over 

This is why we should distinguish transient from non-transient errors now, and also employ exponential backoff. Let's make sure to add an Issue to do it if we don't do it now.

Reply:
yeah, that is why I added the TODO, will create a ticket for it. :)

Reply:
https://github.com/numaproj-labs/numaplane/issues/59